jupyter/sum-to-zero/sum_to_zero_evaluation.qmd

```diff
@@ -0,0 +1,1558 @@
+of demographic predictors with any number of categories.
+
+* The specified test sensitivity and specificity
+
+In order to fit this model, we need to put a sum-to-zero constraint on the categorical variables.
+
+
+### The Stan model
+
+
+The full model is in file `stan/binomial_4_preds_ozs.stan`.
+It provides an estimate of the true prevalence based on binary tests with
+a given (or unknown) test sensitivity and specificity as follows.
+
+```stan
+transformed parameters {
+  // true prevalence
+  vector[N] p = inv_logit(beta_0 + beta_sex * sex_c + beta_age[age]
+			  + beta_eth[eth] + beta_edu[edu]);
+  // incorporate test sensitivity and specificity.
+  vector[N] p_sample = p * sens + (1 - p) * (1 - spec);
+}
+model {
+  pos_tests ~ binomial(tests, p_sample);  // likelihood
+  ...
+```
+
+To constrain the group-level parameters `age`, `eth`, and `edu`,
+we use the `sum_to_zero_vector`.
+
+```stan
+parameters {
+  real beta_0;
+  real beta_sex;
+  real<lower=0> sigma_age, sigma_eth, sigma_edu;
+  sum_to_zero_vector[N_age] beta_age;
+  sum_to_zero_vector[N_eth] beta_eth;
+  sum_to_zero_vector[N_edu] beta_edu;
+}
+```
+
+In order to put a standard normal prior on `beta_age`, `beta_eth`, and `beta_edu`,
+we need to scale the variance, as suggested above.
+The scaling factors are pre-computed in the `transformed data` block,
+and applied as part of the prior.
+
+```stan
+transformed data {
+  // scaling factors for marginal variances of sum_to_zero_vectors
+  real s_age = sqrt(N_age * inv(N_age - 1));
+  real s_eth = sqrt(N_eth * inv(N_eth - 1));
+  real s_edu = sqrt(N_edu * inv(N_edu - 1));
+}
+  ...
+model {
+  ...
+  // centered parameterization
+  // scale normal priors on sum_to_zero_vectors
+  beta_age ~ normal(0, s_age * sigma_age);
+  beta_eth ~ normal(0, s_eth * sigma_eth);
+  beta_edu ~ normal(0, s_edu * sigma_edu);
+}
+```
+
+
+### The data-generating program
+
+To investigate the predictive behavior of this model at different timepoints in a pandemic,
+we have written a data-generating program to create datasets given the
+baseline disease prevalence, test specificity and sensitivity,
+the specified total number of diagnostic tests.
+
+In the `generated quantities` block we use Stan's PRNG functions to populate
+the true weights for the categorical coefficient vectors, and the relative percentages
+of per-category observations.
+Then we use a set of nested loops to generate the data for each demographic,
+using the PRNG equivalent of the model likelihood.
+
+* Because the modeled data `pos_tests` is generated according to the Stan model's likelihood,
+the model is a priori well-specified with respect to the data.
+
+* Because the true parameters are defined in the `generated quantities` block,
+each sample provides a datasets from a different set of regression covariates
+and with different amounts of per-demographic data.
+
+The full data-generating program is in file `stan/gen_binomial_4_preds.stan`.
```

Comment:
(assuming this will end up on e.g. the stan website) should these file references be a link?